### ISSUE TITLE
dt master crashing when using color calibration module

### ISSUE THREAD

**Issue by jenshannoschwalm** [darktable-org/darktable#7528]
**Describe the bug**
dt crashes when using the "color calibration" module

**To Reproduce**
1. Open an image in darkroom
2. Use either one of the sliders in the "grey tab" or select on of the b&w film simulations
3. dt immediately becomes unresponsive and crashes after a short time

**Expected behavior**
not crashing :-)

**Platform**
 - Darktable master
 - OS: Fedora 33, Intel
 - no cl on this machine

**Additional context**
starting on console gives various hints

```
Using cmatrix codepath
malloc(): unaligned tcache chunk detected

(darktable:74518): Gtk-WARNING **: 23:06:12.110: Failed to set text '<markup>' from markup due to error parsing markup: Error on line 1 char 28: Document ended unexpectedly with elements still open ? ?markup? was the last element opened
Abgebrochen (Speicherabzug geschrieben)
```

or
```
(darktable:74817): Gtk-WARNING **: 23:09:08.096: Failed to set text 'R\xb6\u0004\u0008!\u007f' from markup due to error parsing markup: Error on line 1 char 26: Invalid UTF-8 encoded text in name ? not valid ?color ca\xd0\u0008?

(darktable:74817): Gtk-WARNING **: 23:09:08.096: Failed to set text 'R\xb6\u0004\u0008!\u007f' from markup due to error parsing markup: Error on line 1 char 19: Invalid UTF-8 encoded text in name ? not valid ?\xd0\u0008?
malloc_consolidate(): unaligned fastbin chunk detected
Abgebrochen (Speicherabzug geschrieben)
```

**Comment by MStraeten** [comment]
reproduced  a crash when changing color temperature slider (also gamut slider) a couple of times in sequence.
OSX current master darktable 3.5.0+229~gf755cfc28
lldb output:
```
Using cmatrix codepath
Using cmatrix codepath
Using cmatrix codepath
Using cmatrix codepath
darktable(47437,0x70000ec30000) malloc: *** error for object 0x11d4da0e0: pointer being freed was not allocated
darktable(47437,0x70000ec30000) malloc: *** set a breakpoint in malloc_error_break to debug
Process 47437 stopped
* thread #16, name = 'worker res 0', stop reason = signal SIGABRT
    frame #0: 0x00007fff20324462 libsystem_kernel.dylib`__pthread_kill + 10
libsystem_kernel.dylib`__pthread_kill:
->  0x7fff20324462 <+10>: jae    0x7fff2032446c            ; <+20>
    0x7fff20324464 <+12>: movq   %rax, %rdi
    0x7fff20324467 <+15>: jmp    0x7fff2031e6a1            ; cerror_nocancel
    0x7fff2032446c <+20>: retq
Target 0: (darktable) stopped.
(lldb) up
frame #1: 0x00007fff20352610 libsystem_pthread.dylib`pthread_kill + 263
libsystem_pthread.dylib`pthread_kill:
->  0x7fff20352610 <+263>: movl   %eax, %r15d
    0x7fff20352613 <+266>: cmpl   $-0x1, %eax
    0x7fff20352616 <+269>: jne    0x7fff20352660            ; <+343>
    0x7fff20352618 <+271>: movq   %gs:0x8, %rax
(lldb)
frame #2: 0x00007fff202a5720 libsystem_c.dylib`abort + 120
libsystem_c.dylib`abort:
->  0x7fff202a5720 <+120>: movl   $0x2710, %edi             ; imm = 0x2710
    0x7fff202a5725 <+125>: callq  0x7fff20252270            ; usleep$NOCANCEL
    0x7fff202a572a <+130>: callq  0x7fff202a572f            ; __abort

libsystem_c.dylib`__abort:
    0x7fff202a572f <+0>:   pushq  %rbp
(lldb)
frame #3: 0x00007fff20186430 libsystem_malloc.dylib`malloc_vreport + 548
libsystem_malloc.dylib`has_default_zone0:
->  0x7fff20186430 <+0>:  movq   0x6882e361(%rip), %rcx    ; malloc_zones
    0x7fff20186437 <+7>:  testq  %rcx, %rcx
    0x7fff2018643a <+10>: je     0x7fff2018644c            ; <+28>
    0x7fff2018643c <+12>: movq   0x6882e53d(%rip), %rdx    ; initial_default_zone
(lldb)
frame #4: 0x00007fff201894c8 libsystem_malloc.dylib`malloc_report + 151
libsystem_malloc.dylib`malloc_report:
->  0x7fff201894c8 <+151>: movq   0x5fec1559(%rip), %rax    ; (void *)0x00007fff889b3d70: __stack_chk_guard
    0x7fff201894cf <+158>: movq   (%rax), %rax
    0x7fff201894d2 <+161>: cmpq   -0x8(%rbp), %rax
    0x7fff201894d6 <+165>: jne    0x7fff201894e1            ; <+176>
(lldb)
frame #5: 0x00000001009a2f67 libgtk-3.0.dylib`gtk_widget_set_name + 78
libgtk-3.0.dylib`gtk_widget_set_name:
->  0x1009a2f67 <+78>: movq   %r14, 0x28(%rbx)
    0x1009a2f6b <+82>: movq   0x48(%rbx), %rdi
    0x1009a2f6f <+86>: testq  %rdi, %rdi
    0x1009a2f72 <+89>: je     0x1009a2f80               ; <+103>
(lldb)
libdarktable.dylib was compiled with optimization - stepping may behave oddly; variables may not be available.
frame #6: 0x000000010025a20d libdarktable.dylib`_iop_panel_label(lab=0x000000014d62a330, module=0x00000001114e6200) at imageop.c:1268:3 [opt]
   1265
   1266	static void _iop_panel_label(GtkWidget *lab, dt_iop_module_t *module)
   1267	{
-> 1268	  gtk_widget_set_name(lab, "iop-panel-label");
   1269	  char *module_name = dt_history_item_get_name_html(module);
   1270	  gchar *label = g_strdup_printf("%s",
   1271	                                 (module->has_trouble && module->enabled)

```

**Comment by MStraeten** [comment]
bisecting gave:
```
3c27a9457b08c30eb6baf40bc86c0d3afc968dc2 is the first bad commit

    add function to verify piece->colors and set module's trouble flag if… (#7478)
```

**Comment by johnny-bit** [comment]
Missing or bad copy?

Pinging @ralfbrown for this interesting bug caused by #7478 :)

**Comment by TurboGit** [comment]
@ralfbrown : I think all the dt_iop_have_required_input_format() must lock mutex on g->lock.

**Comment by TurboGit** [comment]
I mean the call to `dt_iop_have_required_input_format()` but be done inside a critical section where `g->lock` is locked.

**Comment by ralfbrown** [comment]
OK, here's where I'm really missing inheritance -- without a base class, we'll have to pass g->lock into each invocation of `dt_iop_have_required_input_format` instead of being able to just reference `module->gui_data` as the base class containing the lock....

**Comment by ralfbrown** [comment]
See #7546 for a fix that grew much larger....

**Comment by chhil** [comment]
WSL Ubuntu 20.04
Current master


Created a new instance of color calibration module (instance 1). Was playing around in the grey tab by varying the  RGB sliders. DT window disappeared.
Had the following in the shell that started DT.

(darktable:17378): Gtk-WARNING **: 12:28:11.777: Failed to set text '\xd0;\u0004\xbcK\u007f' from markup due to error parsing markup: Error on line 1 char 27: Invalid UTF-8 encoded text in name — not valid “\xd0;\u0004\xbcK\u007f”
Segmentation fault (core dumped)

Log file created but empty.

Ran dt again and did the same test again and was reproducible.

This time I got the following on the console.
```
munmap_chunk(): invalid pointer
Aborted (core dumped)
```
This time no log file created, not even a file with the version in it.
